pdsClient/pdsClient.go

```diff
@@ -1,110 +1,113 @@
-        return result, err
-    }
-    err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
-    return result, err
+	var result *WriteRecordResponse
+	path := c.Host + "/" + PDSServiceBasePath + "/records"
+	request, err := createRequest("POST", path, params)
+	if err != nil {
+		return result, err
+	}
+	err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
+	return result, err
 }
 
 // ListRecords returns a list of records using any filters provided as params, and error (if any).
 func (c *E3dbPDSClient) ListRecords(ctx context.Context, params ListRecordsRequest) (*ListRecordsResult, error) {
-    var result *ListRecordsResult
-    var buf bytes.Buffer
-    err := json.NewEncoder(&buf).Encode(&params)
-    if err != nil {
-        return result, err
-    }
-    request, err := http.NewRequest("POST", c.Host+"/"+PDSServiceBasePath+"/search", &buf)
-    if err != nil {
-        return result, err
-    }
-    err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
-    return result, err
+	var result *ListRecordsResult
+	path := c.Host + "/" + PDSServiceBasePath + "/search"
+	request, err := createRequest("POST", path, params)
+	if err != nil {
+		return result, err
+	}
+	err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
+	return result, err
+}
+
+// ProxyListRecords returns a list of records using any filters provided as params, and error (if any).
+func (c *E3dbPDSClient) ProxyListRecords(ctx context.Context, authToken string, params ListRecordsRequest) (*ListRecordsResult, error) {
+	var result *ListRecordsResult
+	path := c.Host + "/" + PDSServiceBasePath + "/search"
+	request, err := createRequest("POST", path, params)
+	if err != nil {
+		return result, err
+	}
+	err = e3dbClients.MakeProxiedUserCall(ctx, authToken, request, &result)
+	return result, err
 }
 
 // InternalGetRecord attempts to get a record using an internal only e3db endpoint, returning fetched record and error (if any).
 func (c *E3dbPDSClient) InternalGetRecord(ctx context.Context, recordID string) (*InternalGetRecordResponse, error) {
-    var result *InternalGetRecordResponse
-    request, err := http.NewRequest("GET", c.Host+"/internal/"+PDSServiceBasePath+"/records/"+recordID, nil)
-    if err != nil {
-        return result, err
-    }
-    err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
-    return result, err
+	var result *InternalGetRecordResponse
+	path := c.Host + "/internal/" + PDSServiceBasePath + "/records/" + recordID
+	request, err := createRequest("GET", path, nil)
+	if err != nil {
+		return result, err
+	}
+	err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
+	return result, err
+}
+
+// createListRecordsRequest isolates duplicate code in creating http search request.
+func createRequest(method string, path string, params interface{}) (*http.Request, error) {
+	var buf bytes.Buffer
```

Comment:
Do you think this refactor obfuscates the functions it takes code out of? @galxy25